openpathsampling/shifting.py

```diff
@@ -0,0 +1,123 @@
+import numpy as np
+import random
+import openpathsampling as paths
+import openpathsampling.engines.toy as toy
+import openpathsampling.high_level.move_strategy as strategies
+from openpathsampling import EngineMover
+from openpathsampling.high_level.move_strategy import MoveStrategy
+from openpathsampling.high_level.move_strategy import levels
+from openpathsampling.shooting import NullShootingSelector
+
+
+class ShiftingMover(EngineMover):
+    """
+    ShiftingMover propagates a path backward and forward by a fixed length
+    and cuts the trajectory in the opposite direction of propagation. The
+    propagation direction is chosen from a random uniform distribution.
+
+    Parameters
+    ----------
+    ensemble : :class:`openpathsampling.Ensemble`
+        Ensemble for this shifting mover
+    engine ::class:`openpathsampling.DynamicsEngine`
+        Engine for the shifting mover
+    shift_length ::int
+        Length of propagation of the trajectory (default is 3)
+    """
+
+    def __init__(self, ensemble, shift_length=3, engine=None):
+    #shift length is a time displacement delT from a symmetric distribution around 0
+        super(ShiftingMover, self).__init__(
+            ensemble=ensemble,
+            target_ensemble=ensemble,
+            selector=NullShootingSelector(), #fix
+	    engine=engine
+        )
+
+        self.shift_length = shift_length
+
+    @property
+    def direction(self):
+        return 'bidirectional' # pragma: no cover
+
+
+    def _run(self, trajectory , shooting_index):
+
+        shift = random.choice([-1,1]*self.shift_length)
+
+        if shift > 0: #forward shift condition
+            #print self.shift_length
+            #print 'Initial trajectory', trajectory[1:]
+
+            chopped_traj = trajectory[self.shift_length:] #chop out first shift_length frames
+
+            #generate new trial part of path starting from the end of the chopped traj for fwd shift
+            self.engine.current_snapshot = chopped_traj[-1]  # pick last frame
+            right_part = self.engine.generate_n_frames(self.shift_length) #gen time steps in future to add to
+                                                                          #chopped traj
+            details = {"Forward Shift":self.shift_length}
+
```

Comment:
More a suggestion and question @dwhswenson : Would is make sense to split the details into either
1. `{'shift_direction': ..., 'shift_lenght': }` or even
2. `{'shift`: positive/negative int}`

This way automatic analysis is easier to handle. The current approach would require a `if 'Forward Shift' in x.details`